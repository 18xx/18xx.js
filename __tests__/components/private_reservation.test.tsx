import * as React from 'react';
import * as renderer from 'react-test-renderer';

import Hexagon from '../../src/hexagon';

import PrivateReservation, {
  PrivateReservationProps
} from '../../src/components/private_reservation';

describe('PrivateReservation', () => {
  describe('render', () => {
    const props: PrivateReservationProps = {
      hexagon: new Hexagon('north-south'),
      name: 'C&A',
    };

    describe('when orientation is east-west', () => {
      it('renders an svg', () => {
        const subject: React.ReactElement<PrivateReservation> = (
          <PrivateReservation {...props} />
        );

        expect(renderer.create(subject)).toMatchSnapshot();
      });
    });

    describe('when orientation is north-south', () => {
      it('renders an svg', () => {
        const subject: React.ReactElement<PrivateReservation> = (
          <PrivateReservation {...props} hexagon={new Hexagon('east-west')} />
        );

        expect(renderer.create(subject)).toMatchSnapshot();
      });
    });
  });
});
