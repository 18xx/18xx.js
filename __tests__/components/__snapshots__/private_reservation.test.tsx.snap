// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateReservation render when orientation is east-west renders an svg 1`] = `
<g>
  <circle
    cx={20}
    cy={40.42562584220407}
    r={4}
  />
  <circle
    cx={20}
    cy={70.42562584220407}
    r={4}
  />
  <line
    stroke="black"
    strokeWidth={2}
    x1={20}
    x2={20}
    y1={40.42562584220407}
    y2={70.42562584220407}
  />
  <text
    fill="black"
    fontSize={8}
    textAnchor="start"
    x={24}
    y={57.42562584220407}
  >
    C&A
  </text>
</g>
`;

exports[`PrivateReservation render when orientation is north-south renders an svg 1`] = `
<g>
  <circle
    cx={10}
    cy={49}
    r={4}
  />
  <circle
    cx={10}
    cy={79}
    r={4}
  />
  <line
    stroke="black"
    strokeWidth={2}
    x1={10}
    x2={10}
    y1={49}
    y2={79}
  />
  <text
    fill="black"
    fontSize={8}
    textAnchor="start"
    x={2}
    y={94}
  >
    C&A
  </text>
</g>
`;
